docbook/reference/en/en-US/modules/Reactive.xml

```diff
@@ -1,63 +1,810 @@
+
+    <para>
+       <classname>stage1</classname> returns "foo", <classname>stage2</classname> returns "bar", and
+       <classname>stage3</classname>, which runs when both <classname>stage1</classname> and <classname>stage2</classname>
+       have completed, returns the concatenation of "foo" and "bar". Note that, in this example, an explict
+       <classname>ExecutorService</classname> is provided for asynchronous processing.
+    </para>
+</sect1>
+
+<sect1>
+<title>CompletionStage in JAX-RS</title>
+
+    <para>
+       On the client side, JAX-RS 2.1 mandates an implementation of the interface
+       <classname>javax.ws.rs.client.CompletionStageRxInvoker</classname>:
+    </para>
+    
+<programlisting>
+public interface CompletionStageRxInvoker extends RxInvoker&lt;CompletionStage&gt; {
+
+    @Override
+    public CompletionStage&lt;Response&gt; get();
+
+    @Override
+    public &lt;T&gt; CompletionStage&lt;T&gt; get(Class&lt;T&gt; responseType);
+
+    @Override
+    public &lt;T&gt; CompletionStage&lt;T&gt; get(GenericType&lt;T&gt; responseType);
+    ...
+</programlisting>
+
+    <para>
+       That is, there are invocation methods for the standard HTTP verbs, just as in the standard 
+       <classname>javax.ws.rs.client.SyncInvoker</classname>. A <classname>CompletionStageRxInvoker</classname>
+       is obtained by calling <methodname>rx()</methodname> on a
+       <classname>javax.ws.rs.client.Builder</classname>, which extends <classname>SyncInvoker</classname>.
+       For example,
+    </para>
+    
+<programlisting>
+CompletionStageRxInvoker invoker = client.target(generateURL("/get/string")).request().rx(CompletionStageRxInvoker.class);
+CompletionStage&lt;Response&gt; stage = invoker.get();
+Response response = stage.toCompletableFuture().get();
+System.out.println("result: " + response.readEntity(String.class));
+</programlisting>
+
+    <para>
+       or
+    </para>
+    
+<programlisting>
+CompletionStageRxInvoker invoker = client.target(generateURL("/get/string")).request().rx(CompletionStageRxInvoker.class);
+CompletionStage&lt;String&gt; stage = invoker.get(String.class);
+String s = stage.toCompletableFuture().get();
+System.out.println("result: " + s);
+</programlisting>
+
+    <para>
+       On the server side, the JAX-RS 2.1 specification requires support for resource methods with return type
+       <classname>CompletionStage&lt;T&gt;</classname>. For example,
+    </para>
+    
+<programlisting>
+@GET
+@Path("get/async")
+public CompletionStage&lt;String&gt; longRunningOpAsync() {
+   CompletableFuture&lt;String&gt; cs = new CompletableFuture&lt;&gt;();
+   executor.submit(
+      new Runnable() {
+         public void run() {
+            executeLongRunningOp();
+            cs.complete("Hello async world!");
+         }
+      });
+   return cs;
+}
+</programlisting>
+
+    <para>
+       The way to think about <methodname>logRunningOpAsync()</methodname> is that the method is asynchronously

```

Comment:
Typo, s/logRunningOpAsync/longRunningOpAsync/g

Reply:
@marekkopecky, got it. Thanks.